src/yaml-language-support/schema-utils.ts

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+import { KubernetesGroupKindNode, KubernetesResourceObjectBase } from "./schema-models";
+
+/**
+ * An utility function for splitting text into lines.
+ */
+export function splitLines(text: string): string[] {
+    return text.split(/\r\n?|\n/);
+} 
+
+/**
+ * A very easy utility function for decides whether the text is support to be a kubernetes yaml file, further
+ * implementations is needed to be more precise on the result rather than the regex test.
+ */
+export function isKubernetesYamlFile(text: string): boolean {
+    return !!(/\bkind\s*:/g.exec(text) ||
+        /\bapiVersion\s*:/g.exec(text));
+}
+
+/**
+ * An utility function for get apiVersion and kind from 'x-kubernetes-group-version-kind' node in kubernetes schema.
```

Comment:
I don't think the documentation needs to state that it's a utility function - we know it's a function from the keyword, and that it's a utility from the module name *grin*.  Consider just "Gets apiVersion and kind from..."